src/DIRAC/FrameworkSystem/scripts/dirac_login.py

```diff
@@ -32,206 +49,326 @@
         if not result["OK"]:
             return result
 
-        if self.proxy:
-            os.environ["DIRAC_USE_ACCESS_TOKEN"] = "False"
+        if self.result == "proxy":
+            self.outputFile = self.outputFile or Locations.getDefaultProxyLocation()
             # Save new proxy certificate
-            result = writeToProxyFile(idpObj.token["proxy"].encode("UTF-8"), self.proxyLoc)
+            result = writeToProxyFile(idpObj.token["proxy"].encode("UTF-8"), self.outputFile)
             if not result["OK"]:
                 return result
-            gLogger.notice("Proxy is saved to %s." % self.proxyLoc)
+            gLogger.notice(f"Proxy is saved to {self.outputFile}.")
         else:
-            os.environ["DIRAC_USE_ACCESS_TOKEN"] = "True"
             # Revoke old tokens from token file
-            if os.path.isfile(tokenFile):
-                result = readTokenFromFile(tokenFile)
+            self.outputFile = getTokenFileLocation(self.outputFile)
+            if os.path.isfile(self.outputFile):
+                result = readTokenFromFile(self.outputFile)
                 if not result["OK"]:
                     gLogger.error(result["Message"])
                 elif result["Value"]:
                     oldToken = result["Value"]
                     for tokenType in ["access_token", "refresh_token"]:
                         result = idpObj.revokeToken(oldToken[tokenType], tokenType)
                         if result["OK"]:
-                            gLogger.notice("%s is revoked from" % tokenType, tokenFile)
+                            gLogger.notice(f"{tokenType} is revoked from", self.outputFile)
                         else:
                             gLogger.error(result["Message"])
 
             # Save new tokens to token file
-            result = writeTokenDictToTokenFile(idpObj.token, tokenFile)
+            result = writeTokenDictToTokenFile(idpObj.token, self.outputFile)
             if not result["OK"]:
                 return result
-            tokenFile = result["Value"]
-            gLogger.notice("New token is saved to %s." % tokenFile)
+            self.outputFile = result["Value"]
+            gLogger.notice(f"New token is saved to {self.outputFile}.")
 
             if not DIRAC.gConfig.getValue("/DIRAC/Security/Authorization/issuer"):
                 gLogger.notice("To continue use token you need to add /DIRAC/Security/Authorization/issuer option.")
                 if not self.issuer:
                     DIRAC.exit(1)
                 DIRAC.gConfig.setOptionValue("/DIRAC/Security/Authorization/issuer", self.issuer)
 
-        # Try to get user information
+            # Try to get user authorization information from token
+            result = readTokenFromFile(self.outputFile)
+            if not result["OK"]:
+                return result
+            gLogger.notice(result["Value"].getInfoAsString())
+
+        return S_OK()
+
+    def loginWithCertificate(self):
```

Comment:
it's a slightly simplified version of what's going on in `dirac-proxy-init`.

Reply:
In my opinion, dirac-login should not be too complicated, it should provide the user with a proxy or access token that corresponds to the group.
I don't think we need additional flags like --rfc or --nocs here, no? --VOMS should be added based on group settings, in my opinion.

In any case, each flag must be understood by DIRAC AS (ie as an additional scope) so that the user can obtain the same proxy, both from the local certificate and through DIRAC AS without changing the flags.

Q:
What flags must be for a proxy, and which are redundant for 8.0? do we need everything?

Reply:
Flags `dirac-proxy-init`:
```
Usage:
  dirac-proxy-init [options] ...

Options:
  -v  --valid <value>          : Valid HH:MM for the proxy. By default is 24 hours
  -g  --group <value>          : DIRAC Group to embed in the proxy
  -b  --strength <value>       : Set the proxy strength in bytes
  -l  --limited                : Generate a limited proxy
  -t  --strict                 : Fail on each error. Treat warnings as errors.
  -S  --summary                : Enable summary output when generating proxy
  -C  --Cert <value>           : File to use as user certificate
  -K  --Key <value>            : File to use as user key
  -u  --out <value>            : File to write as proxy
  -x  --nocs                   : Disable CS check
  -p  --pwstdin                : Get passwd from stdin
  -i  --version                : Print version
  -j  --noclockcheck           : Disable checking if time is ok
  -r  --rfc                    : Create an RFC proxy, true by default, deprecated flag
  -L  --legacy                 : Create a legacy non-RFC proxy
  -N  --no-upload              : Do not upload a long lived proxy to the ProxyManager
  -M  --VOMS                   : Add voms extension
  ```
  
Flags `dirac-login`
```
Usage:
  dirac-login [options] ... group scope [scope]

Options:
  -T  --lifetime <value>       : set access lifetime in a hours
  -O  --save-output <value>    : where to save the authorization result(e.g: proxy or tokens). By default we will try to find a standard place.
  -I  --issuer <value>         : set issuer.
  -   --use-certificate        : in case you want to generate a proxy using a certificate. By default.
  -   --use-diracas            : in case you want to authorize with DIRAC Authorization Server.
  -C  --certificate <value>    : user certificate location
  -K  --key <value>            : user key location
  -   --proxy                  : return proxy in case of successful authorization
  -   --token                  : return tokens in case of successful authorization
  -   --status                 : print user authorization status

Arguments:
  group:  select a DIRAC group for authorization, can be determined later. (optional)
  scope:  scope to add to authorization request. (optional)
```

Reply:
My vision is which flags should stay or go and why. What do you think about it?
```
  -v  --valid <value>          : STAY as [T, lifetime]
  -g  --group <value>          : STAY as first argument
  -b  --strength <value>       : GONE, is it not a very special parameter as for the user? Can it be standard for this command?
  -l  --limited                : GONE, do users use this to start a working session with DIRAC?
  -t  --strict                 : GONE, does the user need it? Is there a real case when a user wants to use it?
  -S  --summary                : GONE, IMHO after the command is finished there should be some standard output, if the user does not want it there should be a --quite flag for all commands.
  -C  --Cert <value>           : STAY as [C, certificate]
  -K  --Key <value>            : STAY as [K, key]
  -u  --out <value>            : STAY as [O, save-output]
  -x  --nocs                   : GONE, the essence of this command (roughly speaking) to authorize the user and initialize a working session with DIRAC. This is not possible without accessing the DIRAC server. For what case it may be necessary, given that it is no longer required during DIRAC installation and configure?
  -p  --pwstdin                : Get passwd from stdin
  -i  --version                : GONE, IMHO it should be general flag for all commands
  -j  --noclockcheck           : GONE, if time is OK then check will be OK, no?
  -r  --rfc                    : GONE, deprecated flag
  -L  --legacy                 : GONE, deprecated flag also?
  -N  --no-upload              : GONE, we need to upload a long lived proxy to the ProxyManager to async tasks. Do I need to keep this flag for some cases? 
  -M  --VOMS                   : GONE, IMHO this should be decided on the server side, depending on the group
```

Reply:
I agree with your list with the exception of "--nocs", which I would keep.

Reply:
>   -x  --nocs                   : GONE, the essence of this command (roughly speaking) to authorize the user and initialize a working session with DIRAC. This is not possible without accessing the DIRAC server. For what case it may be necessary, given that it is no longer required during DIRAC installation and configure?

What is not longer required during DIRAC Installation and configure ? Before running `dirac-configure`, we need a proxy, and obviously this one time is ran with the `--cs` flag
Same for the ` --no-upload`

For VOMS, although I see some potential use cases, it's true that it is more of an expert level stuff, so probably not worth it here

Reply:
`dirac-configure` already has logic for that as I known. So you don't need to call `dirac-proxy-init` or `dirac-login` before `dirac-configure`.

Reply:
`dirac-configure` has the logic, but it still uses `--cs --no-upload --pwstdin` under the hood, so they need to be there

Reply:
I think that in this specific case it is more correct to use simply `X509Chain` a class for `dirac-configure`: #5623

Reply:
I don't think `X509Chain` should be used there. It's not really a class meant to be widespread. 
I really see nothing wrong with keeping these options in `dirac-proxy-init`. They are useful, and there are cases when I want to get a proxy without talking to the CS, independently of `dirac-configure`

Reply:
>I don't think X509Chain should be used there. It's not really a class meant to be widespread.

But why? As far as I can see, it is already used in many places, including `dirac-proxy-init`. 

>I really see nothing wrong with keeping these options in dirac-proxy-init. They are useful, and there are cases when I want to get a proxy without talking to the CS, independently of dirac-configure

That is, the real case is to create a proxy for some needs, ie not to interact with the DIRAC server, right? If so then this is not the case for `dirac-login`. I think the right solution might be to leave `dirac-proxy-init` unchanged for this specific case and #5623.

Reply:
> there are cases when I want to get a proxy without talking to the CS

Can you give an example to understand how common this can be?

Reply:
> > there are cases when I want to get a proxy without talking to the CS

> Can you give an example to understand how common this can be?

The case is when you need a proxy in order to talk to the CS.

Reply:
>>>there are cases when I want to get a proxy without talking to the CS

>>Can you give an example to understand how common this can be?

>The case is when you need a proxy in order to talk to the CS.

In this case, you do `dirac-login` and get the desired proxy, during the creation will be synchronized with `CS`. AFAIK `dirac-proxy-init` works the same way.

Reply:
so, `dirac-login` works without the need to talk to a CS service? 
The "--nocs" flag is there to avoid the chicken-and-egg problem.

Reply:
As far as I understand, this is the only question left to merge this PR, so I'll just leave `--nocs`. And I suggest continuing the discussion on dirac-configure elsewhere, in #5623 for example, or wherever you suggest.